Index: components/driver/source/mesh/runtime_partition_lfric_mod.f90
===================================================================
--- components/driver/source/mesh/runtime_partition_lfric_mod.f90	(revision 53845)
+++ components/driver/source/mesh/runtime_partition_lfric_mod.f90	(working copy)
@@ -85,35 +85,6 @@
 
   procedure(partitioner_interface), intent(out), pointer :: partitioner_ptr
 
-  integer(i_def) :: panel_xproc, panel_yproc
-
-  integer :: panel_decomposition
-
-  call partitioning%get_value( 'panel_decomposition', panel_decomposition )
-
-  select case (panel_decomposition)
-
-  case ( panel_decomposition_auto )
-    decomposition = auto_decomposition_type()
-
-  case ( panel_decomposition_row )
-    decomposition = row_decomposition_type()
-
-  case ( panel_decomposition_column )
-    decomposition = column_decomposition_type()
-
-  case ( panel_decomposition_custom )
-    call partitioning%get_value( 'panel_xproc', panel_xproc )
-    call partitioning%get_value( 'panel_yproc', panel_yproc )
-    decomposition = custom_decomposition_type( num_xprocs=panel_xproc, num_yprocs=panel_yproc )
-
-  case default
-    ! Not clear it's possible to still error at this point but no harm in checking
-    call log_event( "Missing entry for panel decomposition, "// &
-                    "specify 'auto' if unsure.", LOG_LEVEL_ERROR )
-
-  end select
-
   call get_partition_strategy(mesh_selection, total_ranks, partitioner_ptr)
 
 end subroutine get_partition_parameters_nml
